cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -513,8 +543,13 @@ class device_row_comparator {
   device_row_comparator(Nullate check_nulls,
                         table_device_view lhs,
                         table_device_view rhs,
-                        null_equality nulls_are_equal = null_equality::EQUAL) noexcept
-    : lhs{lhs}, rhs{rhs}, check_nulls{check_nulls}, nulls_are_equal{nulls_are_equal}
+                        null_equality nulls_are_equal = null_equality::EQUAL,
+                        nan_equality nans_are_equal   = nan_equality::ALL_EQUAL) noexcept
```

Comment:
The NaN configuration is a template parameter for the lex comparator but a runtime configuration for equality? It should be consistent.

Reply:
It gets nasty because of the behavior outlined in #4760 that I'm trying to replicate here. The boolean config triggers a mirroring of the operator (line 269) that allows the `NaN` to `NaN` comparison to give a negative result (e.g. `<` operator will return `weak_ordering::GREATER`).

I can make the equality configuration a template, but will remove the need for the runtime configuration and the comparator structure will diverge a bit (which doesn't seem like a big deal now that I type this out...)

Reply:
See https://github.com/rapidsai/cudf/pull/10870#discussion_r874317958

I think this will be infinitely more flexible and pleasant.